plugin/pkg/auth/authenticator/token/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,169 @@
+//
+// All secrets must be of type "bootstrap.kubernetes.io/token". An example secret:
+//
+//     apiVersion: v1
+//     kind: Secret
+//     metadata:
+//       # Name is arbitrary and isn't used for authentication purposes.
+//       name: bootstrap-token-( token id )
+//       namespace: kube-system
+//     # Only secrets of this type will be evaluated.
+//     type: bootstrap.kubernetes.io/token
+//     data:
+//       token-secret: ( private part of token )
+//       token-id: ( token id )
+//       # Required key usage.
+//       usage-bootstrap-authentication: true
+//       # May also contain an expiry.
+//
+// Tokens are expected to be of the form:
+//
+//     ( token-id ).( token-secret )
+//
+func (t *TokenAuthenticator) AuthenticateToken(token string) (user.Info, bool, error) {
+	tokenID, tokenSecret, err := parseToken(token)
+	if err != nil {
+		// Token isn't of the correct form, ignore it.
+		return nil, false, nil
+	}
+
+	secrets, err := t.lister.List(labels.Everything())
+	if err != nil {
+		return nil, false, err
+	}
+
+	for _, secret := range secrets {
+		if string(secret.Type) != string(bootstrapapi.SecretTypeBootstrapToken) || secret.Data == nil {
+			continue
+		}
+
+		ts := getSecretString(secret, bootstrapapi.BootstrapTokenSecretKey)
+		if ts != tokenSecret {
+			continue
+		}
+
+		id := getSecretString(secret, bootstrapapi.BootstrapTokenIDKey)
+		if id != tokenID {
+			continue
+		}
+
+		if isSecretExpired(secret) {
+			continue
+		}
+
+		if getSecretString(secret, bootstrapapi.BootstrapTokenUsageAuthentication) != "true" {
+			glog.V(3).Infof("Bearer token matching bootstrap Secret %s/%s not marked %s=true.",
+				secret.Namespace, secret.Name, bootstrapapi.BootstrapTokenUsageAuthentication)
+			continue
+		}
+
+		return &user.DefaultInfo{
+			Name:   bootstrapapi.BootstrapUserPrefix + string(id),
+			Groups: []string{bootstrapapi.BootstrapGroup},
+		}, true, nil
+	}
+	return nil, false, nil
+}
+
+// Copied from k8s.io/kubernetes/pkg/bootstrap/api
+func getSecretString(secret *api.Secret, key string) string {
+	if secret.Data == nil {
+		return ""
+	}
+	if val, ok := secret.Data[key]; ok {
+		return string(val)
+	}
+	return ""
+}
+
+// Copied from k8s.io/kubernetes/pkg/bootstrap/api
+func isSecretExpired(secret *api.Secret) bool {
+	expiration := getSecretString(secret, bootstrapapi.BootstrapTokenExpirationKey)
+	if len(expiration) > 0 {
+		expTime, err2 := time.Parse(time.RFC3339, expiration)
+		if err2 != nil {
+			glog.V(3).Infof("Unparseable expiration time (%s) in %s/%s Secret: %v. Treating as expired.",
+				expiration, secret.Namespace, secret.Name, err2)
+			return true
+		}
+		if time.Now().After(expTime) {
+			glog.V(3).Infof("Expired bootstrap token in %s/%s Secret: %v",
+				secret.Namespace, secret.Name, expiration)
+			return true
+		}
+	}
+	return false
+}
+
+// Copied from kubernetes/cmd/kubeadm/app/util/token
+
+var (
+	tokenRegexpString = "^([a-z0-9]{6})\\.([a-z0-9]{16})$"
```

Comment:
@luxas the current regexp in `kubernetes/cmd/kubeadm/app/util/token` actually uses `:` as the separator. I copied this here but switched it to use `.` as the separator.